tsdb/db.go

```diff
@@ -1094,11 +1101,8 @@ func (db *DB) Compact() (returnErr error) {
 		)
 	}
 
-	if lastBlockMaxt != math.MinInt64 {
-		// The head was compacted, so we compact OOO head as well.
-		if err := db.compactOOOHead(); err != nil {
-			return errors.Wrap(err, "compact ooo head")
-		}
+	if err := db.compactOOOHead(); err != nil {
```

Comment:
This will now do OOO compaction every minute because this function is called every minute. We should be removing it from here and putting it inside run() instead.